CONTRIBUTING.md

```diff
@@ -1,51 +1,193 @@
+have encountered something that you believe is a bug, please follow these
+steps:
+
+1. First of all, make sure you are running the latest version of manim.  If
+   not, update your version and try again.
+
+2. Make sure to search for other users who may have had similar issues in the
+   past.  Search the repository's issues page (don't forget to search closed
+   issues), bring it up on Discord, use sites like StackOverflow, and exercise
+   your best Google practices.  If you can't find anything helpful, then go to
+   the next step.
+
+3. Make sure you can reproduce the issue, i.e. that you have some code that
+   illustrates the bug **every time** (or at least most of the time) it is
+   executed.
+   
+4. Make sure to clarify what behavior you expected, and how the actual behavior
+   differs from your expectation.
+
+5. Make sure to gather information about your environment, such as your
+   operating system, python version, and any stack traces that the code may
+   have generated (if applicable).
+   
+4. Please open an issue only after you have gathered this information.  When
+   submitting an issue, make sure to follow the template (this is the default
+   text you are shown when first opening the 'New Issue' page).  A community
+   member will (hopefully) respond and start a conversation to address the
+   issue.
+   
+7. Please give the community a reasonable amount of time before asking again,
+   or insisting on your issue.  Keep in mind that everyone is a volunteer.  If
+   you wait for a reasonable amount of time and you receive no response, feel
+   free to ask again.
+
+
+## Contributing code
+
+Many ways of contributing will involve writing, reading, testing, or
+refactoring code.  As our repository is a Fork of [Manim by
+3b1b](https://github.com/3b1b/manim), contributing in this way can be a bit
+confusing.  Here is a short guide on how to do it.
+
+1. First, make a fork of this repository.  This creates your own copy of the
+   whole codebase on GitHub.
+
+2. Clone the repository you forked using the command below (bash/zsh).
 
-## General instructions
+   ```sh
+     git clone <your-fork-url>
+     cd manim
+   ```
 
-As this is a Fork of [Manim by 3b1b](https://github.com/3b1b/manim), contributing can be a bit confusing. Because of that, here is a small guide on it. 
+	Note: do not `git clone` the original ManimCommunity repository.  You must
+    clone your own fork.  After this step, there are three different
+    repositories to keep track of: the original ManimCommunity repo, your own
+    fork of it, and your local repository.
 
-1. First, make a fork of this repository.
-2. Then clone the repository you forked using the command below (bash/zsh).
+3. You have to make `git` be aware of the three repositories, by entering the
+   commands below.
 
- ```sh
-   git clone <your-fork-url>
-   cd manim
- ```
+   ```sh
+   git remote add upstream https://github.com/ManimCommunity/manim.git
+   git pull upstream master
```

Comment:
**Issue**: It should be `git fetch` and not `git pull`. `git pull` will create a merge commit which is unnecessary hear and could cause problems.
```suggestion
   git fetch upstream
```